### PR TITLE
Fix bug where users with privileged access can't download learning objects

### PR DESCRIPTION
**Pull request by nrwinner**
**Problem:** With the addition of collections into the access groups (`reviewer` changing to `reviewer@collection`) the current implementation client side identified reviewers and curators as regular users and this denied them download access.

**Solution:** Before iterating the accessGroups array, remove references to collections entirely.

**Future:** A more robust solution should be implemented wherein there are collection checks client side to prevent a user downloading an object from a collection they're not privileged in. This isn't absolutely necessary at this time since the service will only send down objects that the current user is authorized to see. @gshaw1997 @davidnso can you confirm this last bit?

### CODE REVIEW

File: src/app/core/auth.service.ts [hunk 1]
```diff
@@ -299,14 +300,14 @@ export class AuthService {
   }
 
   /**
-   * Identifies if the current logged in user has reviewer priviledges.
+   * Identifies if the current logged in user has reviewer privileges.
```
Comment (DonSeannelly) [inline src/app/core/auth.service.ts]: ❤️

File: src/app/core/auth.service.ts [hunk 2]
```diff
@@ -319,12 +320,17 @@ export class AuthService {
     if (!this.user['accessGroups']) {
       return;
     }
-    if (this.user['accessGroups'].includes('admin')) {
+
+    const groups = this.user['accessGroups'].map(x => x.split('@')[0]);
```
Comment (DonSeannelly) [inline src/app/core/auth.service.ts]: If this is the solution we want to go with, can we add a comment here to explain this as it was explained in the PR description?

File: src/app/core/auth.service.ts (hunk 1 above)
Comment (nrwinner) [reply src/app/core/auth.service.ts]: The funny part is I misspelled the same word in my commit message

**Comment by gshaw1997** [comment]
> **Problem:** With the addition of collections into the access groups (`reviewer` changing to `reviewer@collection`) the current implementation client side identified reviewers and curators as regular users and this denied them download access.
> 
> **Solution:** Before iterating the accessGroups array, remove references to collections entirely.
> 
> **Future:** A more robust solution should be implemented wherein there are collection checks client side to prevent a user downloading an object from a collection they're not privileged in. This isn't absolutely necessary at this time since the service will only send down objects that the current user is authorized to see. @gshaw1997 @davidnso can you confirm this last bit?

@nrwinner Yes, they will only see objects they have access to.

**Pull request merged**